### ISSUE TITLE
Performance degrades with distance from training data across all splitting strategies

### ISSUE THREAD

**Issue by srijitseal** [srijitseal/polaris#10]
## Summary

Training XGBoost (fixed architecture) on ECFP4 + full RDKit 2D descriptors (~200, with dimorphite_dl protonation at assay pH) across all 9 endpoints and 3 splitting strategies (cluster, time, target). Targets are log-transformed for non-LogD endpoints, matching the competition protocol. Evaluated with competition metrics: MAE, R², Spearman ρ, RAE, MA-RAE.

### MA-RAE by splitting strategy

| Strategy | MA-RAE | Interpretation |
|----------|--------|----------------|
| Cluster | **0.726** | Standard 5-fold CV, ~80:20 per fold |
| Time | 0.837 | Expanding window (50:50 → 80:20), temporal ordering |
| Target | 6.927 | By design creates impossible extrapolation tasks |

### Per-endpoint metrics (cluster-split, mean across 5 folds)

| Endpoint | MAE | R² | Spearman ρ | RAE |
|----------|-----|-----|-----------|-----|
| LogD | 0.502 | 0.659 | 0.824 | 0.557 |
| Caco-2 Efflux | 0.237 | 0.483 | 0.654 | 0.672 |
| MBPB | 0.246 | 0.453 | 0.696 | 0.700 |
| KSOL | 0.422 | 0.383 | 0.556 | 0.743 |
| MPPB | 0.279 | 0.377 | 0.632 | 0.771 |
| MLM CLint | 0.418 | 0.370 | 0.592 | 0.750 |
| Caco-2 Papp | 0.279 | 0.367 | 0.664 | 0.756 |
| MGMB | 0.219 | 0.363 | 0.607 | 0.755 |
| HLM CLint | 0.379 | 0.273 | 0.562 | 0.830 |

### Per-endpoint metrics (time-split, mean across 4 folds)

| Endpoint | MAE | R² | Spearman ρ | RAE |
|----------|-----|-----|-----------|-----|
| LogD | 0.525 | 0.608 | 0.777 | 0.609 |
| MBPB | 0.262 | 0.348 | 0.664 | 0.790 |
| Caco-2 Papp | 0.282 | 0.290 | 0.595 | 0.789 |
| KSOL | 0.476 | 0.271 | 0.493 | 0.823 |
| HLM CLint | 0.388 | 0.256 | 0.548 | 0.850 |
| MPPB | 0.293 | 0.227 | 0.533 | 0.864 |
| Caco-2 Efflux | 0.288 | 0.198 | 0.590 | 0.788 |
| MLM CLint | 0.466 | 0.106 | 0.473 | 0.909 |
| MGMB | 0.260 | -0.156 | 0.506 | 1.107 |

### Structural distance degradation

| Endpoint | Cluster | Time | Target |
|----------|---------|------|--------|
| LogD | **2.89x** | 2.71x | 1.25x |
| MGMB | **2.89x** | 1.76x | 0.92x |
| HLM CLint | **2.75x** | 1.63x | 1.12x |
| MPPB | **2.33x** | 2.07x | 0.94x |
| MLM CLint | **2.21x** | 2.62x | 1.14x |
| KSOL | **1.94x** | 1.53x | 1.43x |
| MBPB | **1.93x** | 1.67x | 0.70x |
| Caco-2 Papp | **1.06x** | 1.00x | 1.04x |
| Caco-2 Efflux | 1.03x | 1.11x | 0.36x |

### Interpretation

- **Cluster-split** (MA-RAE 0.726) gives the best overall performance since each fold sees structurally diverse training data. R² ranges from 0.27 (HLM CLint) to 0.66 (LogD). Structural degradation is 2-3x for most endpoints.
- **Time-split** (MA-RAE 0.837) is worse because early folds have small training sets (expanding window starts at 50:50). R² drops below 0 for MGMB where training data is smallest (86 molecules at fold 0).
- **Target-split** (MA-RAE 6.927) produces catastrophically negative R² across all endpoints. This is by design — folds are ordered by target value, so test molecules systematically fall outside the training distribution in target space. Spearman correlations collapse to near-zero (~0.01–0.07), confirming the model cannot rank molecules it hasn't seen in the relevant value range.
- **Caco-2 endpoints show minimal structural degradation** (1.0–1.1x for cluster) — suggesting less structural diversity between clusters or better generalization for permeability.
- **Comparison to 2.08 baseline** (MA-RAE 0.820 on competition split): cluster-split CV gives a more optimistic estimate (0.726) than the held-out test set, which is expected — the competition split likely contains more structurally novel molecules.

## Model

- XGBRegressor: fixed config (n_estimators=1000, max_depth=6, lr=0.1, subsample=0.8, colsample_bytree=0.4, min_child_weight=5, gamma=1.0, reg_alpha=0.1, reg_lambda=1.5, tree_method="hist")
- Features: 2048-bit ECFP4 + ~192 RDKit 2D descriptors (scaled), computed on dimorphite_dl-protonated SMILES at assay pH (7.4 for most, 6.5 for Caco-2)
- Targets: log10(clip(x, 1e-10) + 1) for all endpoints except LogD

## Plots

<img width="2078" height="877" alt="Image" src="https://github.com/user-attachments/assets/2f7f1e98-786c-401e-abd8-51f8311aba4d" />
<img width="2678" height="921" alt="Image" src="https://github.com/user-attachments/assets/7580e581-d538-4ca6-9883-769be26f8f13" />
<img width="2078" height="877" alt="Image" src="https://github.com/user-attachments/assets/203fbda8-2ad3-449c-a993-ad4974462aa5" />
<img width="2078" height="877" alt="Image" src="https://github.com/user-attachments/assets/bd4c8a0c-fdb7-42cf-a05f-6f6a973bb9c1" />
<img width="2678" height="2299" alt="Image" src="https://github.com/user-attachments/assets/364ac601-0ca5-4a5e-b96e-7ceb5338be14" />
<img width="2676" height="2298" alt="Image" src="https://github.com/user-attachments/assets/41f409ff-bf75-415c-9839-69cbf0e02d93" />

## Reproduce

```bash
pixi run -e cheminformatics python notebooks/2.07-seal-performance-distance.py
```

## Source

`notebooks/2.07-seal-performance-distance.py`

**Comment by srijitseal** [comment]
**Update (2026-03-19): Re-run with chirality-aware ECFP4 (`useChirality=True`, PR #16).**

Performance degradation pattern unchanged. Degradation ratios (farthest/closest bin RMSE) remain consistent across all strategies. Conclusion unchanged.

**Figures refreshed** in `data/processed/2.07-seal-performance-distance/`.

<img width="2078" height="877" alt="Image" src="https://github.com/user-attachments/assets/18d9f714-2c3c-42db-a6e6-1715fe7645db" />
<img width="2678" height="921" alt="Image" src="https://github.com/user-attachments/assets/2cfe8939-de72-4e33-a7a8-b66cf2bddb3f" />
<img width="2078" height="877" alt="Image" src="https://github.com/user-attachments/assets/2eb4f578-bc6e-41fb-8a54-6ed276e8fe91" />
<img width="2078" height="877" alt="Image" src="https://github.com/user-attachments/assets/82e042f6-27b0-40ed-87eb-3c6b4081ca16" />
<img width="2678" height="2299" alt="Image" src="https://github.com/user-attachments/assets/369bed5d-3182-4ce0-8e37-0f0bae302c8c" />
<img width="2676" height="2298" alt="Image" src="https://github.com/user-attachments/assets/e380c966-f75f-4f87-869e-bf2a1547c384" />

**Comment by srijitseal** [comment]
## CheMeleon update (2.07 performance over distance)

Re-ran the three splitting strategies (cluster, time, target) with the CheMeleon foundation model under the same Optuna-style protocol — single fine-tuned model per fold, no ensemble.

**Verdict: trends match XGBoost. Cluster < time ≪ target ordering preserved; structural degradation under cluster-split persists. One mechanistic claim from the XGBoost analysis no longer holds — see Caco-2 below.**

### MA-RAE by strategy

| Strategy | XGBoost | CheMeleon | Δ |
|----------|---------|-----------|---|
| Cluster | 0.673 | **0.625** | −0.048 |
| Time | 0.772 | **0.728** | −0.044 |
| Target | 6.106 | 5.259 | −0.847 |

Same ranking. Foundation model lifts the floor on every strategy but does not collapse the gap between strategies. Target-split remains catastrophic (R² −57 vs −81 — both useless).

### Cluster-split per-endpoint (mean across 5 folds)

| Endpoint | R² XGB | R² CM | RAE XGB | RAE CM |
|----------|--------|-------|---------|--------|
| LogD | 0.722 | 0.830 | 0.502 | 0.376 |
| MBPB | 0.595 | 0.681 | 0.599 | 0.502 |
| MPPB | 0.541 | 0.589 | 0.653 | 0.597 |
| Caco-2 Efflux | 0.545 | 0.520 | 0.611 | 0.612 |
| Caco-2 Papp | 0.444 | 0.260 | 0.690 | 0.776 |
| KSOL | 0.437 | 0.457 | 0.703 | 0.600 |
| MGMB | 0.355 | 0.408 | 0.720 | 0.646 |
| MLM CLint | 0.401 | 0.427 | 0.745 | 0.726 |
| HLM CLint | 0.237 | 0.297 | 0.831 | 0.793 |

CheMeleon improves R² on 7/9 endpoints under cluster-split; worse on Caco-2 Papp (0.44 → 0.26) and Caco-2 Efflux (0.55 → 0.52).

### Structural degradation (cluster-split, RMSE farthest / RMSE closest bin)

| Endpoint | XGB | CM |
|----------|-----|-----|
| LogD | 3.21× | 3.54× |
| HLM CLint | 2.98× | **4.30×** |
| MPPB | 2.14× | 2.63× |
| MLM CLint | 2.08× | 1.92× |
| KSOL | 1.90× | 1.34× |
| MBPB | 1.82× | 2.54× |
| MGMB | 1.91× | 2.32× |
| Caco-2 Papp | 0.96× | **2.91×** |
| Caco-2 Efflux | 0.97× | **1.60×** |

### Trend match — important nuance

The XGBoost write-up explained Caco-2 robustness (0.95–0.98×) as *"passive permeability is governed by global properties (size, lipophilicity, H-bonding) captured by RDKit 2D descriptors."* **That mechanistic claim does not survive under CheMeleon** — Caco-2 Papp degrades 2.91× and Caco-2 Efflux 1.60× when the model has no access to the global descriptors. The flat XGBoost curve was a feature of the input representation, not a property of the endpoints. With a graph-based foundation model, structural degradation appears for these endpoints too.

For all other endpoints the qualitative picture is unchanged: 1.3–4.3× degradation from closest to farthest distance bin under cluster-split, and the worst-degraded endpoints (LogD, HLM CLint) remain the worst-degraded under CheMeleon.

### Source

- `data/processed/2.07-seal-performance-distance/chemeleon/per_fold_performance.csv`
- `data/processed/2.07-seal-performance-distance/combined/metrics_comparison.csv`
- Reproduce: `pixi run -e cheminformatics python notebooks/2.07-seal-performance-distance.py --model chemeleon`

**Comment by srijitseal** [comment]
## Refreshed numbers (XGBoost rerun + CheMeleon)

The issue body has been re-synced against current artifacts under `data/processed/2.07-seal-performance-distance/xgboost/`. Strategy-level drift:

| Strategy | Old MA-RAE | Refreshed |
|----------|------------|-----------|
| Cluster | 0.673 | 0.673 |
| Time | 0.771 | **0.772** |
| Target | 6.118 | **6.106** |

Per-endpoint cluster-split degradation ratios refreshed (LogD 3.60× → **3.21×**, HLM CLint 3.04× → **2.98×**, MGMB 1.87× → **1.91×**, MBPB 1.90× → **1.82×**, etc.). The Caco-2 robustness narrative was also revised: the flat XGBoost curve (Papp 0.96×, Efflux 0.97×) does *not* hold under CheMeleon (Papp **2.91×**, Efflux **1.60×**), so it's a representation effect of ECFP4 + RDKit 2D, not a property of the endpoints.

CheMeleon comparison (MA-RAE: cluster 0.625, time 0.728, target 5.259) stands — see #issuecomment-4376596586.

**Comment by srijitseal** [comment]
## Degradation ratios updated (2026-05-17)

Re-ran notebook 2.07 from scratch (deleted stale predictions parquet and regenerated). The previous `performance_distance_summary.csv` was inconsistent with the predictions parquet — they were from different model states due to pred_cache staleness.

### Updated XGBoost cluster-split degradation ratios

| Endpoint | Old | New |
|----------|-----|-----|
| LogD | 3.21× | **3.60×** |
| HLM CLint | 2.98× | **3.04×** |
| KSOL | 1.90× | **2.01×** |
| MGMB | 1.91× | **1.87×** |
| MBPB | 1.82× | **1.90×** |
| Caco-2 Papp | 0.96× | **0.98×** |
| Caco-2 Efflux | 0.97× | **0.95×** |
| MPPB | 2.14× | 2.14× (unchanged) |
| MLM CLint | 2.08× | 2.08× (unchanged) |

CheMeleon ratios confirmed unchanged (4.30×, 2.91×, 1.60× etc.).

### New supplementary analysis: distance metric comparison (notebook 2.17)

Added comparison of Tanimoto vs RDKit 2D descriptor Euclidean distance as degradation axis. Key finding: descriptor distance shows sharper degradation for 7/9 endpoints (notably KSOL 1.31→2.26×, Caco-2 Papp 0.97→1.51×), while Tanimoto is sharper for LogD and HLM CLint. Neither metric uniformly dominates.

Plot: `data/processed/2.17-seal-distance-metric-comparison/distance_metric_comparison.png`

**Comment by srijitseal** [comment]
**Update: Combined figure created**

Combined Figure S1 (baseline_combined_panel.png) added to the combined/ folder, showing XGBoost and CheMeleon baseline results side-by-side.
